### PR TITLE
Define index mappings from collection metadata

### PR DESCRIPTION
**Pull request by leplatrem**
Ref #8 

This is another approach to #36.

Since conversion from JSONSchema to ES mapping seems fragile. Let's take another curve and do something less smart but a lot simpler: read an explicit attribute `index:schema` in the collection metadata. 

@glasserc @n1k0 @Natim @magopian Thoughts?

### CODE REVIEW

File: tests/test_elasticsearch.py [hunk 1]
```diff
@@ -162,3 +162,89 @@ def test_search_is_not_allowed_if_only_read_on_certain_records(self):
...
+    def test_index_has_mapping_if_collection_has_schema(self):
+        mapping = self.get_mapping("bid", "cid")
+        assert mapping == {
+            "properties": {
+                "id": {"type": "keyword"},
+                "last_modified": {"type": "long"},
+                "build": {
+                    "properties": {
+                        "date": {"type": "date", "format": "strict_date"},
+                        "id": {"type": "keyword"}
```
Comment (Natim) [inline tests/test_elasticsearch.py]: type: string

File: tests/test_elasticsearch.py [hunk 2]
```diff
@@ -162,3 +162,89 @@ def test_search_is_not_allowed_if_only_read_on_certain_records(self):
...
+
+    def test_index_has_mapping_if_collection_has_schema(self):
+        mapping = self.get_mapping("bid", "cid")
+        assert mapping == {
+            "properties": {
+                "id": {"type": "keyword"},
+                "last_modified": {"type": "long"},
+                "build": {
+                    "properties": {
+                        "date": {"type": "date", "format": "strict_date"},
```
Comment (Natim) [inline tests/test_elasticsearch.py]: format: dateOptionalTime

File: tests/test_elasticsearch.py [hunk 3]
```diff
@@ -162,3 +162,89 @@ def test_search_is_not_allowed_if_only_read_on_certain_records(self):
...
+        indexer = self.app.app.registry.indexer
+        indexname = indexer.indexname(bucket_id, collection_id)
+        index_mapping = indexer.client.indices.get_mapping(indexname)
+        return index_mapping[indexname]["mappings"][indexname]
+
+    def test_index_has_mapping_if_collection_has_schema(self):
+        mapping = self.get_mapping("bid", "cid")
+        assert mapping == {
+            "properties": {
+                "id": {"type": "keyword"},
```
Comment (Natim) [inline tests/test_elasticsearch.py]: type: string

Comment (leplatrem) [inline tests/test_elasticsearch.py]: `The string field is unsupported for indexes created in 5.x in favor of the text and keyword fields. `

**Pull request merged**